jsonkeytransformer.go

```diff
@@ -11,30 +12,51 @@ import (
 // fully support ECMA-404 (for instance, Elasticsearch 2.x does
 // not allow periods in key names, which ECMA-404 does)
 type JSONKeyTransformer struct {
+	old      string
+	new      string
```

Comment:
Being slightly nitpicky here, I usually prefer things like `oldVal` and `newVal`. I generally stay away from using keywords as data members.